idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/AddLateInitFactory.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.quickfix.fixes
+
+import org.jetbrains.kotlin.idea.fir.api.fixes.diagnosticFixFactory
+import org.jetbrains.kotlin.idea.frontend.api.fir.diagnostics.KtFirDiagnostic
+import org.jetbrains.kotlin.idea.frontend.api.types.isPrimitive
+import org.jetbrains.kotlin.idea.quickfix.AddModifierFix
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtProperty
+
+object AddLateInitFactory {
+
+    @OptIn(ExperimentalStdlibApi::class)
+    val addLateInitFactory = diagnosticFixFactory<KtFirDiagnostic.MustBeInitializedOrBeAbstract> { diagnostic ->
+        val property: KtProperty = diagnostic.psi
```

Comment:
I think it aids readability when seeing this code outside the IDE. Is there a recommendation that we should skip any optional type annotations?

Reply:
I suggested this because you also don't have the explicit type in `addAccessorsToUninitializedProperty`, and I think it reads just fine since the variable name is `property` and you can see it's getting the PSI element with `.psi`.

Reply:
Thanks for pointing it out! Let me add the type in `addAccessorsToUninitializedProperty`.

I prefer to have explicit type here because it helps me to confirm the type is indeed `KtProperty`. Without it, the reader may think that the static type is a `PsiElement`, though we are expecting it to be a `KtProperty` at runtime. The presence of this type annotation helps me to rule out that possibility, therefore increasing my confidence of understanding what this code is doing.